File: psup assignment/cpp/q4.cpp
#include <iostream>
#include <string>
#include <cctype>
int main() {
    int num;
    std::cout << "\n--- Program 4: Even Number Check ---\n";
    std::cout << "Enter a number: ";
    std::cin >> num;
    if (num % 2 == 0) {
        std::cout << "The number is even." << std::endl;
    }
    return 0;
}

File: psup assignment/java/q4.java
import java.util.*;
class EvenCheck {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter a number: ");
        int num = input.nextInt();
        if (num % 2 == 0) {
            System.out.println("The number is even.");
        }
        input.close();
    }
}